air/src/air/trace_info.rs

```diff
@@ -80,7 +77,6 @@ impl TraceInfo {
     pub fn new_multi_segment(
         main_segment_width: usize,
         aux_segment_width: usize,
-        num_aux_segment_rands: usize,
         trace_length: usize,
         trace_meta: Vec<u8>,
     ) -> Self {
```

Comment:
The struct that implements `Prover` would be responsible for that. As noted in [this comment](https://github.com/facebook/winterfell/pull/271#discussion_r1589000554), the `RescueRapsProver` is an example of that: it has a variable `num_rand_elements` (renamed from `aux_segment_width`).

However the `Air` no longer knows about it, so the `AuxTraceVerifier` also needs to "hardcode" the same constant... Will look into it more

Reply:
As a temporary solution (see [comment](https://github.com/facebook/winterfell/pull/271#discussion_r1589078479)), I think this shouldn't be a problem, since the number of random elements should always be known by both the prover and verifier as part of the problem definition? I believe we'd be able to clean this up properly if we brought the GKR implementation to winterfell.